packages/blazor-workspace/NimbleBlazor/CONTRIBUTING.md

```diff
@@ -87,11 +87,14 @@ Test Project: `NimbleBlazor.Tests.Acceptance`
+The `NimbleBlazor.Tests.Acceptance` project starts a local Blazor Web App which serves Razor pages that host the Nimble components. Then, xUnit-based acceptance tests start a Chromium instance using [Playwright](https://playwright.dev/), load those Razor pages, and interact with them. The majority of the tests use the `InteractiveServer` render mode, but the project also supports the Interactive Web Assembly render mode (and static server-side rendering mode) for tests.
 
-To add a new acceptance test:
-- Add a new Razor file that uses that component in the `Pages` subfolder, with the name `[ComponentName][FunctionalityUnderTest].razor`, e.g. `DialogOpenAndClose.razor`. Add any necessary code to initialize the component in a `@code` section in the same file. If you'll interact with the component as the test runs, you may need to add other Nimble components like buttons to trigger new actions on your component under test.
-- In the `Tests` subfolder, add a new class `[ComponentName]Tests.cs` if it doesn't already exist. Add a new test method in that class. Load your Razor file with the `NewPageForRouteAsync(routeName)` method. Using the Playwright APIs, interact with the components on the page, and make assertions about the state of the component under test.
+To add a new acceptance test (with the Interactive Server render mode):
+- Add a new Razor file that uses that component in the `Pages.InteractiveServer` subfolder, with the name `[ComponentName][FunctionalityUnderTest].razor`, e.g. `DialogOpenAndClose.razor`.
+  - Add any necessary code to initialize the component in a `@code` section in the same file. If you'll interact with the component as the test runs, you may need to add other Nimble components like buttons to trigger new actions on your component under test.
+  - Ensure that your Razor file has `@rendermode RenderMode.InteractiveServer` in the top section.
+  - Specify the page's URL route via a `@page` directive - for example, `@page "/InteractiveServer/DialogOpenAndClose"`.
+- In the `Tests.InteractiveServer` subfolder, add a new class `[ComponentName]Tests.cs` if it doesn't already exist. Add a new test method in that class. Load your Razor file with the `NewPageForRouteAsync(routeName)` method. Using the Playwright APIs, interact with the components on the page, and make assertions about the state of the component under test.
 
 See the existing acceptance tests for examples of using the Playwright APIs. Additionally, see [Getting Started with Playwright Tests (Skyline End2EndTests)](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/End2EndTests/Getting%20Started%20with%20Playwright%20Tests.md&_a=preview) and the [Playwright .NET docs on writing tests](https://playwright.dev/dotnet/docs/writing-tests).
 
```

Comment:
Do the vscode commands a bit further down in this file need to be updated to include the hybrid project?

I hadn't run these in ages so don't know if they worked before, but I got this when I ran the wasm example on macOS. After following the instructions in the error message both apps worked fine for me (besides my other comment about minor bugs). I don't think this requires any action in the PR, just letting you know that .net8 works on macOS!
 
```
      The ASP.NET Core developer certificate is in an invalid state. To fix this issue, run 'dotnet dev-certs https --clean' and 'dotnet dev-certs https' to remove all existing ASP.NET Core development certificates and create a new untrusted developer certificate. Use 'dotnet dev-certs https --trust' to trust the new certificate.
fail: Microsoft.Extensions.Hosting.Internal.Host[11]
      Hosting failed to start
      System.InvalidOperationException: Unable to configure HTTPS endpoint. No server certificate was specified, and the default developer certificate could not be found or is out of date.
      To generate a developer certificate run 'dotnet dev-certs https'. To trust the certificate (Windows and macOS only) run 'dotnet dev-certs https --trust'.
      For more information on configuring HTTPS see https://go.microsoft.com/fwlink/?linkid=848054.
         at Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps(ListenOptions listenOptions, Action`1 configureOptions)
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(ListenOptions[] listenOptions, AddressBindContext context, Func`2 useHttps, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
         at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
         at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
```

Reply:
Maybe the CONTRIBUTING should link out to this topic: https://learn.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-8.0&tabs=visual-studio%2Clinux-sles#trust-the-aspnet-core-https-development-certificate-on-windows-and-macos

Reply:
Added commands for hybrid, update docs to include those commands, linked to doc about the ASP.NET Core certificate stuff

Reply:
I'm guessing this has always been the case, but when I run the hybrid command from my mac I get the error below. It's targeting Windows-specific stuff so I think that makes sense. Longer term maybe we want to rename it to `Demo.Hybrid.Win` and add a `Demo.Hybrid.Mac`, assuming that's supported. But I think it's fine to leave as-is for now.

Re-opening thread for visibility but if that all sounds sane to you, no action needed.

```
You must install or update .NET to run this application.

App: /Users/jesse/Documents/GitHub/nimble/packages/blazor-workspace/Examples/Demo.Hybrid/bin/Debug/net8.0-windows7.0/Demo.Hybrid
Architecture: arm64
Framework: 'Microsoft.WindowsDesktop.App', version '8.0.0' (arm64)
.NET location: /usr/local/share/dotnet

No frameworks were found.

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.WindowsDesktop.App&framework_version=8.0.0&arch=arm64&rid=osx-arm64&os=osx.15
dotnet watch ❌ Exited with error code 150
```

Reply:
Actually, adding a "(Windows only)" next to those commands in CONTRIBUTING couldn't hurt

Reply:
Blazor hybrid should support macos, maybe our target is overly specific? https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/publish/?view=aspnetcore-8.0

Reply:
Added (Windows only) to the docs for the commands.

If we want to investigate getting the hybrid project working on Mac I think that would become a separate work item.